x509/common.go

```diff
@@ -19,6 +22,23 @@ const (
 	factoryCaName string = "Factory-CA"
 )
 
+const (
+	// CRL constants for device CA revokation.
+	// Currently, the API will permanently revoke the device CA on reasons 1-5, 9-10; and ignore reasons 0, 6-8.
+
+	// Revoke the device CA, so that no device client certificates issued by this CA are recognized.
+	// This action is permanent.
+	CrlCaRevoke = 9 // RFC 5280 - privilegeWithdrawn
+
+	// Disable the device CA, so that no new devices can be registered with client certificates issued by this CA.
+	// Devices that were already registered can still connect and operate as normal.
+	// This action can be reverted by CrlCaRenew.
+	CrlCaDisable = 6 // RFC 5280 - certificateHold
+
+	// Renew the previously disabled device CA, so that new device registrations are possible again.
+	CrlCaRenew = 8 // RFC 5280 - removeFromCRL
```

Comment:
I'm not sure that I want to drag this whole baggage right now.
Probably, I will simply define the `CrlCaRevoke` as internal constant inside `x509/golang.go` and change the `CreateCrl` signature to accept a list of serials.

This is here just to show you how we can further improve our functionality going forward.
Although, there is a small chance that after walking a dog on this, you would come up and tell me that this is a must have.

Reply:
@doanac do you think it is worth pursuing this _disable but not revoke device CA_ agenda now or later.

I'm working on the API side ATM, and will scrap this code once I test this end-2-end; unless we make a decision that we need it right now.

Reply:
Disable but not revoke is the feature I'm interested in. I get how CRL is useful for lots of *clients*. But for client certs in a real factory, I don't think its very useful. How often in practice is someone going to want to say, "remove this CA and never trust the potentially hundreds of devices it signed"? If I lose control over my CA, I'd want to:

 1) tell our backend to stop trusting new devices ASAP
 2) look at devices in the backend signed with this CA and try to decide if they are legit or not. If not legit, delete the device

Reply:
Cool, then I see 2 options how to proceed.

The background:
- The _revoke_ feature only affects one Fioctl command and one API endpoint, and is fairly easy to implement.
- The _disable_ feature adds little complexity to the Fioctl on top of the above; but it requires additional changes to registration handlers in the API and device gateway.

So, our options are:
A. Do the _revoke_ part now, and do the _disable_ part in Q1, as I have other priority items for Q4.
B. Do both _revoke_ and _disable_ features now, ETA is a few days end-2-end.

I _believe_ we still have time in reserve for option B. @doanac what do you advice?

Reply:
We could probably argue your other priorities are higher than either of these options. My thinking is "revoke" can be done now - I've done it many times working on AWS and Azure IOT integrations. Its just messy/manual. But our other priorities are for things that can't be done without changes.

That said, "disable" is something we *can't* do. So if you can do disable w/o a lot of disruption - I'm fine.